offchain-message/src/lib.rs

```diff
@@ -245,12 +427,182 @@ impl OffchainMessage {
     }
 }
 
+/// Envelope for off-chain messages with multiple signatures
+/// All signers listed in the message must provide signatures (no threshold logic)
+/// This implements the envelope format from the spec:
+/// | Field | Start offset | Length (bytes) | Description |
+/// | Signature Count | 0x00 | 1 | Number of signatures |
+/// | Signatures | 0x01 | `SIG_COUNT` * 64 | ed25519 signatures |
+/// | Message Preamble | 0x01 + `SIG_COUNT` * 64 | variable | The message preamble |
+/// | Message Body | varies | variable | The message content |
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Envelope {
```

Comment:
I'm not sure whether this is implemented elsewhere and I can just pull it in; will look more later. Any tips appreciated if so.